backend/de.metas.deliveryplanning.base/src/main/java/de/metas/deliveryplanning/process/M_Delivery_Planning_GenerateShortageOverage.java

```diff
@@ -0,0 +1,264 @@
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final DeliveryPlanningId deliveryPlanningId = DeliveryPlanningId.ofRepoId(getRecord_ID());
+
+		final DocTypeQueryBuilder docTypeQuery = DocTypeQuery.builder()
+			.docBaseType(MaterialPhysicalInventory)
+			.adClientId(getClientID().getRepoId())
+			.adOrgId(OrgId.ANY.getRepoId());
+
+		final AdMessageKey adMessageKey;
+		if (p_Qty.signum() > 0)
+		{
+			adMessageKey = MSG_Event_OverageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryOverageDocument);
+		}
+		else if (p_Qty.signum() < 0)
+		{
+			adMessageKey = MSG_Event_ShortageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryShortageDocument);
+		}
+		else
+		{
+			throw new AdempiereException(MSG_ERROR_QUANTITY_ZERO);
+		}
+
+		final I_M_Inventory inventory = generateInventoryDocument(docTypeDAO.getDocTypeId(docTypeQuery.build()), deliveryPlanningId);
+		createUserNotification(inventory, adMessageKey);
+
+		return MSG_OK;
+	}
+
+	private I_M_Inventory generateInventoryDocument(@NonNull final DocTypeId doctypeId, @NonNull final DeliveryPlanningId deliveryPlanningId)
+	{
+		final DeliveryPlanningReceiptInfo receiptInfo = deliveryPlanningService.getReceiptInfo(deliveryPlanningId);
+		final InOutId inOutId = receiptInfo.getReceiptId();
+		Check.assumeNotNull(inOutId, "InOutId shall be set, because of isReceived() check in preconditions");
+
+		final I_M_InOut inOutRecord = load(inOutId, I_M_InOut.class);
+		final List<I_M_InOutLine> inOutLines = inOutDAO.retrieveLines(inOutRecord);
+		if(inOutLines.size() == 0)
+		{
+			throw new AdempiereException("No receipt lines found");
+		}
+		if(inOutLines.size() > 1)
+		{
+			throw new AdempiereException("More than 1 receipt line exists");
+		}
+
+		final I_M_Inventory inventoryRecord = newInstance(I_M_Inventory.class);
+		inventoryRecord.setAD_Org_ID(receiptInfo.getOrgId().getRepoId());
+		inventoryRecord.setC_DocType_ID(doctypeId.getRepoId());
+		inventoryRecord.setM_Warehouse_ID(inOutRecord.getM_Warehouse_ID());
+		inventoryRecord.setMovementDate(inOutRecord.getMovementDate());
+
+		saveRecord(inventoryRecord);
+		logger.trace("Insert inventory - {}", inventoryRecord);
+
+		createInventoryLine(InventoryId.ofRepoId(inventoryRecord.getM_Inventory_ID()), inOutLines.get(0));
+
+		return inventoryRecord;
+	}
+
+	private void createInventoryLine(@NonNull final InventoryId inventoryId, @NonNull final I_M_InOutLine inOutLineRecord)
+	{
+		final TableRecordReference mInOutLineTableRecordReference = TableRecordReference.of(inOutLineRecord);
+		final Set<TableRecordReference> mInOutLineTableRecordReferenceSet = Collections.singleton(mInOutLineTableRecordReference);
+		final ImmutableSetMultimap<TableRecordReference, HuId> huIdsMultimap = huAssignmentDAO.retrieveHUsByRecordRefs(mInOutLineTableRecordReferenceSet);
+		final Collection<HuId> huIds = huIdsMultimap.get(mInOutLineTableRecordReference);
+		final I_M_HU huRecord = load(CollectionUtils.singleElement(huIds), I_M_HU.class);
```

Comment:
you don't need the HUrecord in here, so better pass the HuId to the strategy, and load the record only when you really, really need it

Reply:
i thought it would be better to load it once when the strategy is created instead of every time when `de.metas.handlingunits.inventory.draftlinescreator.ShortageAndOverageStrategy#streamHus()` needs it for `huForInventoryLineFactory.ofHURecordWithQtyAdjustment`

Reply:
before i forget: if you load it, please use `IHandlingUnitsDAO#getById`.